app/api/answers/submit.go

```diff
@@ -74,16 +76,19 @@ func (srv *Service) submit(rw http.ResponseWriter, httpReq *http.Request) servic
 
 	service.MustNotBeError(render.Render(rw, httpReq, service.CreationSuccess(map[string]interface{}{
 		"answer_token": &token.Answer{
-			Answer:         *requestData.Answer,
-			UserID:         requestData.TaskToken.UserID,
-			ItemID:         requestData.TaskToken.ItemID,
-			ItemURL:        requestData.TaskToken.ItemURL,
-			LocalItemID:    requestData.TaskToken.LocalItemID,
-			UserAnswerID:   strconv.FormatInt(userAnswerID, 10),
-			RandomSeed:     requestData.TaskToken.RandomSeed,
-			HintsRequested: hintsInfo.HintsRequested,
-			HintsGiven:     strconv.FormatInt(int64(hintsInfo.HintsCached), 10),
-			AttemptID:      requestData.TaskToken.AttemptID,
+			Answer:          *requestData.Answer,
+			UserID:          requestData.TaskToken.UserID,
+			ItemID:          requestData.TaskToken.ItemID,
+			ItemURL:         requestData.TaskToken.ItemURL,
+			LocalItemID:     requestData.TaskToken.LocalItemID,
+			UserAnswerID:    strconv.FormatInt(userAnswerID, 10),
+			RandomSeed:      requestData.TaskToken.RandomSeed,
+			HintsRequested:  hintsInfo.HintsRequested,
+			HintsGivenCount: strconv.FormatInt(int64(hintsInfo.HintsCached), 10),
+			AttemptID:       requestData.TaskToken.AttemptID,
+			PlatformName:    srv.TokenConfig.PlatformName,
+
+			PrivateKey: srv.TokenConfig.PrivateKey,
```

Comment:
It gives the impression the pv key is given as output. Actually, it is related with the already mentioned "problem" of signing as part of the JSON serialization. `t = token.Answer{...}.sign(pvkey);` with rendering on a separate line would have been clearer. 
Fix it only if you think it is better.

Reply:
added .Sign(pvkey)